Remove DockerHub IPv6 Registry (#7460)
gardener/gardener: docs/proposals/21-ipv6-singlestack-local.md

The `docker.io` registry doesn't support pulling images over IPv6 (see [Beta IPv6 Support on Docker Hub Registry](https://www.docker.com/blog/beta-ipv6-support-on-docker-hub-registry/)).

> Container images from `docker.io` used on shoots and seeds are rewritten to `mirror.gcr.io` which supports dual-stack network access.

Reviewer: Can you be more specific and say that we are hard-coding all docker hub images in the respective `images.yaml` (in g/g and extensions)?

> Container images from `docker.io` used on shoots and seeds are rewritten to `mirror.gcr.io` which supports dual-stack network access.

Reviewer: Can you document this in https://github.com/gardener/gardener/blob/master/docs/usage/ipv6.md as well?
Also, please add a point to https://github.com/gardener/gardener/blob/master/docs/development/component-checklist.md that no images should be used from docker hub because of the well-known rate limits and the missing dual-stack access.

> Container images from `docker.io` used on shoots, seeds and extensions are hard-coded to `mirror.gcr.io` in all respective `images.yaml`. The Google container registry supports dual-stack access and has no rate-limit.

Reviewer 2: Isn't exactly this missing from this PR? I still see images from Docker hub used in https://github.com/gardener/gardener/blob/master/charts/images.yaml :)

Author: yes this is true, since these images are not yet deployed when performing a `make kind-up`, they were not yet addressed.
The next PR, which handles `make gardener-up` will address this an change all images that we currently use from docker.io

> Container images from `docker.io` used on shoots, seeds and extensions are hard-coded to `mirror.gcr.io` in all respective `images.yaml`. The Google container registry supports dual-stack access and has no rate-limit.

Reviewer: ```suggestion
Container images from `docker.io` used in shoots, seeds, and extensions are hard-coded to `mirror.gcr.io` in all respective `images.yaml`. The [Google Mirror](https://cloud.google.com/container-registry/docs/pulling-cached-images) of Docker Hub supports dual-stack access and has no rate limit.
```
